icu4j/main/classes/core/src/com/ibm/icu/impl/number/LongNameHandler.java

```diff
@@ -115,6 +117,25 @@ static void getMeasureData(
             key.append(unit.getSubtype());
         }
 
+        // Grab desired case first, if available. Then grab nominative case to fill
+        // in the gaps.
+        //
+        // TODO(icu-units#138): check that fallback is spec-compliant
+        if (width == UnitWidth.FULL_NAME
+                        && unitDisplayCase != null
+                        && !unitDisplayCase.isEmpty()) {
+            StringBuilder caseKey = new StringBuilder();
+            caseKey.append(key);
+            caseKey.append("/case/");
+            caseKey.append(unitDisplayCase);
+
+            try {
+                resource.getAllItemsWithFallback(caseKey.toString(), sink);
+            } catch (MissingResourceException e) {
+                throw new IllegalArgumentException("No data for unit " + unit + ", width " + width, e);
```

Comment:
This fallback logic is wrong: if there isn't "WIDE" data available for the desired case, it should fall back to not looking for a case. The error message should also include mention of "case", because otherwise it looks like more fundamental data is missing, rather than "only" inflected data.

```suggestion
                throw new IllegalArgumentException("No data for unit " + unit + ", width " + width + ", case " + unitDisplayCase, e);
```

Not a full review yet, I'm just looking into some unit test failures (wanted to check if there's a data problem). I'm seeing 4 erros and 1 failure.

Reply:
This is done.